### PR TITLE
Support option overrides and test skipping

### PR DESCRIPTION
**Pull request by iay**
WIP:

- Reworked README to be clearer, and add documentation for overrides.
- Supports option overrides in general based on `endpoint` name matching.
- Supports `skip: true`.

### CODE REVIEW

**Comment by alexstuart** [comment]
I like this a lot & approve (in both senses of the word).

Two nits:
1. line 57 yout typoed "imlementing"

2. lines 60-62 you write   
> `v09x` represents the current production deployment. Its name reflects that environment's composition: v0.9.x of the Shibboleth Metadata Aggregator, and the Xalan XSL processor included in the classpath.

You're using using x to signify "with Xalan" and also "a patch version". Might be clearer if you don't add the patch version.

**Comment by iay** [comment]
> > `v09x` represents the current production deployment. Its name reflects that environment's composition: v0.9.x of the Shibboleth Metadata Aggregator, and the Xalan XSL processor included in the classpath.
> 
> You're using using x to signify "with Xalan" and also "a patch version". Might be clearer if you don't add the patch version.

Oh, *good* catch. I will just say v0.9 in this case, and fix up the other thing. Cheers.

File: README.md [hunk 1]
```diff
@@ -35,7 +31,50 @@ docker compose exec driver rake validate_all
...
+set of *status metadata* which is returned to the testbed driver.
+
+A test *succeeds* if the returned status metadata matches the
+test's expected results. The test *fails* if any differences
+are detected.
+
+## Endpoints
+
+The `docker compose` deployment currently includes four containers
+implementing metadata validation web services which are referenced
```
Comment (philsmart) [inline README.md:57]: Maybe you could add 'fleet' here, e.g. 'implementing the fleet of metadata validation web services'

File: README.md [hunk 2]
```diff
@@ -35,7 +31,50 @@ docker compose exec driver rake validate_all
...
+inside the testbed as the following named *endpoints*:
+
+- `v09x` represents the current production deployment. Its name reflects
+  that environment's composition: v0.9 of the Shibboleth Metadata
+  Aggregator, and the Xalan XSL processor included in the classpath.
+- `v09` is identical to `v09x` with the exclusion of the Xalan processor.
+  Because this can run neither the old nor new versions of the URL checking
+  code (the old system involves a Xalan extension, while the new system only
```
Comment (philsmart) [inline README.md:65]: I understand this, but I was just wondering if there was a link somewhere which explained what that URL checking code was, or a ticket—something for reference. Although, I guess it is unlikely somebody without the required context would need to understand this.

Comment (iay) [reply README.md:65]: It was my thinking that this was indeed too "inside baseball" to be worth expanding. Most of the documentation around this issue is in the private version of the `ukf-meta` repository, so it's also difficult to link to.

Comment (philsmart) [reply README.md:65]: Yup. That makes sense.

**Pull request merged**